src/vscode-dts/vscode.d.ts

```diff
@@ -946,6 +946,21 @@ declare module 'vscode' {
 		constructor(id: string, color?: ThemeColor);
 	}
 
+	/**
+	 * Represents an icon in the UI. This is either an uri, separate uris for the light- and dark-themes,
+	 * or a {@link ThemeIcon theme icon}.
+	 */
+	export type IconPath = Uri | {
```

Comment:
@mjbvz Let me know if you have an idea for a better name. This isn't overly correct but matches the property names. 

Also there are still a few places where `iconPath: ...` is used/defined but where not this exact shape is supported